commands/update.bee.inc

```diff
@@ -101,3 +119,72 @@ function update_db_bee_callback() {
     bee_message(bt('All pending updates applied.'), 'success');
   }
 }
+
+/**
+ * Command callback: Show, and apply, module and theme updates.
+ *
+ * 2023-03-06 a first attempt at getting this working
+ */
+function update_bee_callback() {
+  global $_bee_backdrop_root;
+  require_once $_bee_backdrop_root . '/core/includes/file.inc';
+  $themes = list_themes();
+  $data = NULL;
+  if ($available = update_get_available(TRUE)) {
+    module_load_include('inc', 'update', 'update.compare');
+    $data = update_calculate_project_data($available);
+  }
+  if ($data != NULL) {
+    foreach ($data as $item) {
+      if ($item['name'] == 'backdrop') {
+        unset($data['backdrop']);
+      }
+      else {
+        if ($item['existing_version'] == $item['latest_version']) {
+          unset($data[$item['name']]);
+        }
+      }
+    }
+  }
+  if ($data != NULL) {
+    bee_message(bt('These are the items being updated.....'));
+    foreach ($data as $item) {
+      bee_message(bt($item['name'] . ' upgrading from ' . $item['existing_version'] .
+        ' to ' . $item['latest_version']));
+    }
+    // TODO Prompt to continue. but it does not show
+    // if (!bee_confirm(bt('Would you like to continue?'), FALSE)) {
+    //    return;
+    // }
+    foreach ($data as $item) {
+      $folder = $_bee_backdrop_root . '/' . $item['project_type'] . 's/' . $item['name'];
+      if (is_dir($folder)) {
+        delete_dir($folder);
+      }
+      download_bee_callback(array('projects' => array($item['name']),), []);
+      echo '';
+    }
+    bee_message(bt('You will probably need to fix the permissions on the new modules'), 'warning');
+  } else {
+    bee_message(bt('No Modules or Themes to Update'));
+  }
+}
+
+/*
+ * Delete a folder recursively
+ * checks for root folder and exits
+ * */
+/**
+ * @param $path - Path of folder to delete
+ * @return bool
+ */
+function delete_dir($path): bool {
+  if (empty($path)) {
+    return false;
+  }
+  $files = array_diff(scandir($path), array('.', '..'));
+  foreach ($files as $file) {
+    (is_dir("$path/$file")) ? delete_dir("$path/$file") : unlink("$path/$file");
+  }
+  return rmdir($path);
+}
```

Comment:
We don't need this.

```suggestion
```

Reply:
@yorkshire-pudding I have applied those suggestions. All look very sensible to me